loadbalancer/etcd/client.go

```diff
@@ -25,27 +30,69 @@ type client struct {
 // The parameter machines needs to be a full URL with schemas.
 // e.g. "http://localhost:4001" will work, but "localhost:4001" will not.
 func NewClient(machines []string, cert, key, caCert string) (Client, error) {
-	var c *etcd.Client
-	var err error
+	var c etcd.KeysAPI
```

Comment:
Presuming you do introduce functional options + an options struct, the optional parameters cert, key, caCert should be removed from the parameter list of the constructor and become functional options. Perhaps all in one go, e.g.

```go
func Certified(cert, key, caCert string) Option {
    return func(o *options) { o.cert, o.key, o.caCert = cert, key, caCert }
}
```